src/compiler/checker.ts

```diff
@@ -13845,7 +13845,9 @@ namespace ts {
                             }
                             break;
                     }
-
+                    if (isFunctionLike(location)) {

```

Comment:
This isn't quite right. Arrow functions shouldn't automatically get an `arguments` in the completion list, nor does it get them in call signatures, etc.

Add the following to `utilities.ts`

```TypeScript
function introducesArgumentsExoticObject(node: Node) {
    switch (decl.kind) {
        SyntaxKind.MethodDeclaration:
        SyntaxKind.MethodSignature:
        SyntaxKind.Constructor:
        SyntaxKind.GetAccessor:
        SyntaxKind.SetAccessor:
        SyntaxKind.FunctionDeclaration:
        SyntaxKind.FunctionExpression:
            return true;
    }
    return false;
}
```

Then use that here. I'd add a test like

```TypeScript
function f() {
    let g = () => /*shouldHaveArguments*/
}

let g = () => /*shouldNotHaveArguments*/
```